app/jobs/runtime/prune_completed_tasks.rb

```diff
@@ -13,10 +13,12 @@ def initialize(cutoff_age_in_days)
         def perform
           logger = Steno.logger('cc.background')
           logger.info('Cleaning up old TaskModel rows')
-
           tasks_to_delete = TaskModel.where(state: prunable_states).where(Sequel.lit('updated_at < ?', cutoff_age))
-          deleted_count   = Database::BatchDelete.new(tasks_to_delete).delete
+          task_labels_to_delete = TaskLabelModel.where(task: tasks_to_delete)
+          deleted_label_count = Database::BatchDelete.new(task_labels_to_delete).delete
```

Comment:
Can you also do this for [Task Annotations](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/app/models/runtime/task_annotation_model.rb), which are the [second component of the CCNG metadata concept](https://v3-apidocs.cloudfoundry.org/version/3.155.0/index.html#annotations)?  I think they could cause a similar error.

Reply:
Added annotation removal [in this commit](https://github.com/cloudfoundry/cloud_controller_ng/pull/3612/commits/c817d92ab222193d5a917dd2341f93166a6ebead)